app/Http/Controllers/AdminAccountController.php

```diff
@@ -46,16 +46,33 @@ public function storeUserByCSV(Request $request){
                 if(trim($row['surname']) == ""){
                     $row['surname'] = null;
                 }
-                if(trim($row['email']) == ""){
-                    $row['email'] = null;
-                }
-                if (!filter_var($row['email'], FILTER_VALIDATE_EMAIL)) {
-                    $row['email'] = null;
+
+                if(key_exists('email', $row)){
+                    if(trim($row['email']) == ""){
+                        $row['email'] = null;
+                    }
+                    if (!filter_var($row['email'], FILTER_VALIDATE_EMAIL)) {
+                        $row['email'] = null;
+                    }
                 }
             } catch (\Exception $e){
                 return redirect()->back()->withErrors("A required column is missing.");
             }
 
+            //Add the hash
+            try {
+                $row['hash'] = UserController::createNewHash();
+            } catch (\Exception $e){
+                return redirect()->back()->withErrors("There was a problem with the User Hash. Please try again");
+            }
+
+            //If there is no email address, it will use the Iserv-Email address
+            if(!key_exists('email', $row) || is_null($row['email'])){
+                //Add the email address Format: max.mustermann@gym-mellendorf.de
+                $row['email'] = $row['first_name'] . "." . $row['surname'] . "@gym-mellendorf.de";
+                $row['email'] = strtolower($row['email']);
```

Comment:
Are there cases where this pattern doesn't apply?
E.g. if two Students are named "Max Mustermann"
Shouldn't there be a fallback so one mail-address belongs to two users?